Keyboard shortcuts + Wrap Up

Larsie13 opened the pull request:
In a nutshell, I added:
- Keyboard shortcuts for the Kanji and Vocab tabs, and for switching between them (the SRS tab does not have any shortcuts yet). Also for the SRS windows. Can you look at it and let me know if the way I did it (especially how I indicated the shortcuts in tooltips) is okay with you?
- The Wrap Up feature in reviews. It looks a little awkward because we need a new button, which really shouldn't be hidden while you haven't submitted an answer yet like the other buttons. And it would look even weirder to put that button on a separate row, all on its own. Maybe I could put it at the top, near the 'Close Review' button. Actually, come to think of it, the 'Close Review' button can really just be replaced with the 'Wrap Up' button, can't it?

Doublevil commented:
There still seems to be an issue when filtering both by meaning and by category. Keyboard shortcuts are cool though.

Larsie13 commented:
Well, the issue with the shared filtering is now fixed. Also, I found another issue regarding filtering myself  where filtering with certain radical combinations didn't work (using WK's radical set, it's Mouth + Butcher where I know it didn't work -- where Butcher is probably called Left/Right Village in other sets).

Larsie13 commented:
This last commit implemented issue #12.
Next, I'll be adding keyboard shortcuts to the SRS page.

After that, I think it's just a matter of cleaning up the new filters (i.e. make the category/WK/JLPT filters look less like they were just thrown in there) and then we're just about ready for a 1.3 release, wouldn't you say?

Doublevil commented:
I just tested a bit and did not see any problem. I was mainly worried about how the ',' separator was handled but it's not an issue after all.
I'd like for the "wrap up" feature to be there for 1.3 though, at least.
Do you want me to merge this request now and have you submit another when you're done with the rest of the shortcuts? Or do you prefer to keep working on this one?

Larsie13 commented:
Yeah, I fixed the filtering issue in the last commit. Also, in case you missed it, the "wrap up" feature is already there; I replaced the old "end session" button. BTW: On WK, a batch is 10 items, meaning that 'Wrap Up' gives you 10 more items there. In Houhou, it's currently 7. Should we make that 10 to be more similar to WK, or should we keep it at 7?
Also, note to self: either change the 'wrapping up...' text to indicate how many items are left before you're done with the review or change the total number of items in the top right once you're wrapping up.

Just leave the merge request open for now; I'll edit the title once the rest of the shortcuts are in there.

And speaking of the separator, the way I understand it, definitions in the EDICT DB have regular commas replaced with the ALT+0130 one, right? I think the only problem with that would be if you wanted to manually add a meaning that contained a comma.

Doublevil commented:
Oh wow, I had never actually reviewed the wrap up feature and it completely slipped out of my mind. I'm sorry!
So, ideally we should still have a way to end the review session immediately. Maybe with a small "x" button on the right of the wrap up one? And yeah it would be nice to have the number of remaining items displayed on the wrap up button once we're in wrap up.

About batch size, I initially set it up to 7 but maybe 10 would be better? I don't really know. If you feel like setting it to 10, I think you already know that but you just have to change the constant. :p I'm okay with that.

And yeah you're right about the commas. I don't consider it a problem not to be able to add meanings containing a comma. That said, I'm still angry at my younger self taking the decision to use commas as a separator. Semicolon would've been so much better. :(

Thank you for your work. :+1:

Larsie13 commented:
Come to think of it, a few weeks ago, the use of commas for separator actually turned out to be a slight problem for me. I was adding 螺 to Houhou (to save you the trouble of looking it up, it's a jinmeiyou meaning 'small, edible, helical fresh-water mollusk' and also used in the word 螺子 (ねじ - screw)). As you might be able to tell by looking at the meaning of the kanji, I initially wasn't aware that it was one meaning; I thought it had three different meanings. So yeah, with that being said, I feel semicolons would work better (especially since that would allow people to directly copy meanings from elsewhere in Houhou without having to change the EDICT semicolons to commas).
I think that might be something to look into for 1.4 or whatever version gets the reverse SRS system, since we'll have to overhaul the SRS database anyway, as you said.

Also, I was thinking about the new filters earlier and got to thinking maybe we should add filters for the other things that you currently can't filter by. For kanji, that would be stroke count, whether it's a Jouyou, Secondary or Jinmeiyou and maybe something about the frequency (not sure about that last one, though). I'll look into that when I'm cleaning up the current filters.

Larsie13 commented:
Aaand... there's the new shortcuts + wrap up item count.
Is it all right with you this way?

Also, the vocab category filter isn't currently working 100%; the ateji/Phonetic category has no vocab, according to Houhou. Probably because I currently only filter by the category of vocab _meanings_, rather than vocab _items_ because I thought that no vocab items had a category. The ateji category does. That should be fixed eventually, but I don't feel like messing around in SQL at the moment, so I'll do that next time (unless you find time and interest to check it out -- the query is at VocabDao line 645).

Larsie13 commented:
I noticed it quite some time ago, actually, but thought at the time that it was just the 'all readings' filter not working properly. But when I saw a name earlier today that I couldn't read and saw somewhere else that it was because the kanji had a bunch of nanori readings, I realized what was going on.

...Now to figure out which kanji to set back to the lowest SRS level because I don't know half of its readings, lol.

Also, now that this is working, I've noticed something else, except I don't know if it's a bug or a feature. On the kanji add/edit page for SRS, the arrow-down button next to the kanji character ignores nanori readings. Bug or feature?

Doublevil commented:
Nanori readings aren't actually important at all (they are garbage most of the time), which is why they are disabled by default and they don't show up in accepted readings when you auto-fill. So yeah this is by design.

Larsie13 commented:
What do you think about obeying the ShowNanori setting and copying the nanori readings for the two or three people that _do_ care about them (like me)?

I certainly wouldn't call them garbage. I mean, if you want to be able to read names, you're gonna need to know the nanori readings at some point (assuming you ever see names other than 山田, 田中 and 佐藤 -- like 恭彦 [やすひこ], which I encountered earlier today and which you won't be able to read if you don't know that 恭 has a nanori of やす).

Doublevil commented:
Yeah sorry that's not exactly what I meant. I just meant that the KANJIDIC2 nanori readings are a bit too exhaustive. Often there are like a dozen or so nanori readings and only a couple are useful. Most dictionaries don't even have the option to show them and I think it's mainly because of that.

Now of course you can add a setting for that if you really want to, but I'd like for it to be distinct from the ShowNanori setting. Showing up the nanori readings for reference can be useful, but having them added to the accepted readings is a different need in my opinion.

Larsie13 commented:
Oh, sorry, I misunderstood, then. Yeah, I've seen kanji with a lot of nanori readings before, too. 一, 大, 上, 日 and 二 are great (horrible?) examples of those.

Actually, thinking about it some more, I think it would be infinitely more useful to be able to select and copy the readings on the kanji page. What do you think? Good idea?

Doublevil commented:
Yeah why not. Maybe a right click context menu on any reading? Or a small icon button right next to the reading category labels?

Larsie13 commented:
As far as I'm concerned, you can merge this pull request. I don't know if you're okay with releasing the current version as 1.3, but aside from a few additions to 'What's New', I think there isn't much left to do.

mrahhal commented:
Will this be merged anytime? Those look like important additions.
I feel like this project isn't being maintained anymore?
